src/servers/http_server_v2.cc

```diff
@@ -0,0 +1,1539 @@
+
+#if 0
+      // Reserve requested space in evbuffer...
+      struct evbuffer_iovec output_iovec;
+      if (evbuffer_reserve_space(evhttp_buffer, byte_size, &output_iovec, 1) !=
+          1) {
+        return TRTSERVER_ErrorNew(
+            TRTSERVER_ERROR_INTERNAL,
+            std::string(
+                "failed to reserve " + std::to_string(byte_size) +
+                " bytes in output tensor buffer")
+                .c_str());
+      }
+
+      if (output_iovec.iov_len < byte_size) {
+        return TRTSERVER_ErrorNew(
+            TRTSERVER_ERROR_INTERNAL,
+            std::string(
+                "reserved " + std::to_string(output_iovec.iov_len) +
+                " bytes in output tensor buffer, need " +
+                std::to_string(byte_size))
+                .c_str());
+      }
+
+      output_iovec.iov_len = byte_size;
+      *buffer = output_iovec.iov_base;
+
+      // Immediately commit the buffer space. We are relying on evbuffer
+      // not to relocate this space. Because we request a contiguous
+      // chunk every time (above by allowing only a single entry in
+      // output_iovec), this seems to be a valid assumption.
+      if (evbuffer_commit_space(evhttp_buffer, &output_iovec, 1) != 0) {
+        *buffer = nullptr;
+        return TRTSERVER_ErrorNew(
+            TRTSERVER_ERROR_INTERNAL,
+            "failed to commit output tensors to output buffer");
+      }
+#endif
+
+      LOG_VERBOSE(1) << "HTTP using buffer for: '" << tensor_name
+                     << "', size: " << byte_size << ", addr: " << *buffer;
+    }
+  }
+
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+HTTPAPIServerV2::ResponseRelease(
+    TRTSERVER_ResponseAllocator* allocator, void* buffer, void* buffer_userp,
+    size_t byte_size, TRTSERVER_Memory_Type memory_type, int64_t memory_type_id)
+{
+  LOG_VERBOSE(1) << "HTTP release: "
+                 << "size " << byte_size << ", addr " << buffer;
+
+  // Don't do anything when releasing a buffer since ResponseAlloc
+  // wrote directly into the response ebvuffer.
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+ReadDataArrayFromJson(
+    const rapidjson::Value& request_input, char** base, size_t* byte_size)
+{
+  const rapidjson::Value& tensor_data = request_input["data"];
+  const rapidjson::Value& shape = request_input["shape"];
+  const char* dtype = request_input["datatype"].GetString();
+
+  // Must be an array
+  if (!tensor_data.IsArray()) {
+    return TRTSERVER_ErrorNew(
+        TRTSERVER_ERROR_INVALID_ARG,
+        "failed to parse request buffer, tensor data must be an array");
+  }
+
+  std::vector<int> shape_vec;
+  int element_cnt = 0;
+  for (rapidjson::SizeType i = 0; i < shape.Size(); i++) {
+    shape_vec.push_back(shape[i].GetInt());
+    if (element_cnt == 0) {
+      element_cnt = shape[i].GetInt();
+    } else {
+      element_cnt *= shape[i].GetInt();
+    }
+  }
+
+  // If flat
+  if (!tensor_data[0].IsArray()) {
+    // BOOL vector is broken
```

Comment:
Why? Also, within Triton bool is an uint8 with value 0 or 1 so you should create a uint8_t vector and initialize it that way.